### DISCUSSION TITLE
Using pre-built Kraken2 / Bracken DB

### DISCUSSION THREAD

**Discussion by seoldh** [bhattlab/phanta#26] [Phanta DBs] [answered]
Hi all,
Thanks for developing such a great tool.
I have found that the taxon i'm looking for does not exist in the Humgut based db. So I just want to use the pre-built db that kraken2 provides (Standard; https://benlangmead.github.io/aws-indexes/k2).
How do I make it compatible with phanta?

**Comment by meenachakra** [comment]
Hello and thank you for your interest in Phanta!

Question - which taxon are you interested in? We are thinking of expanding the prokaryotic portion of our DB soon to include more species, including those that are lowly prevalent in the human gut. 

We could definitely help you make the standard K2 database compatible with Phanta if you wish, but please note that it does not include nearly as many viruses as MGV (please see Fig. 3B of our [paper](https://www.nature.com/articles/s41587-023-01799-4/figures/3)).

**Reply by seoldh** [reply]
Thank you for your reply.
 I am interested in the oral and gastric fluid microbiome, which includes taxa such as T. denticola, H. pylori, etc. that are not in current Phanta DB. I had completely forgotten that MGV is integrated into the DB, and as you mentioned, using standard k2 DB doesn't seem to be a good option.
I am glad to hear that you are working on updating the DB soon. I would like to ask if you are also planning to add oral or gastric fluid microbiomes? and could you provide an estimated timeline for its release date?

**Reply by meenachakra** [reply]
Hi @seoldh , Got it, thank you for the details!
Currently we do not have plans to specifically incorporate oral and gastric fluid microbiomes; we are considering just making a version of the DB with UHGGv2 as the new prokaryotic portion. Unlike HumGut, UHGGv2 is not "filtered" on prevalence of taxa in the human gut.
UHGGv2 does contain some of the taxa you are interested in, such as H. pylori. This database should be released within a few weeks. 
We may also make a more expansive DB but that would be probably in September.

**Reply by meenachakra** [reply, marked as answer]
If you "watch" this repo I think you can get updates on when we make changes (and you can manage the notifications in your settings).

**Comment by gholizade** [comment]
Hi @meenachakra , First, thank you for developing Phanta, which is a very promising workflow. I ran Phanta successfully using the default database (unmasked_db_v1) and integrated_prophages_detector.py which is important to me. I now wish to use Kraken2 database k2_viral_20231009 for Phanta. I have download this database which include Phanta requires like: names.dmp, nodes.dmp, seqid2taxid.map, taxo.k2d, hash.k2d, opts.k2d,  inspect.txt (I renamed to inspect.out),and the library/ folder. When I run the pipeline I hit this error:                                                                     Error in rule filter_kraken:
  jobid: 89
  output: … .krak.report.filtered …
  shell: python …/filter_kraken_reports.py <report> /path/to/k2_viral_20231009  0.01 0.1 …  
The traceback indicates the filtering step failed with a non‑zero exit code. Could you kindly help me to solve the problem?

**Comment by meenachakra** [comment]
Hi! Thanks for your kind words. This page has a list of the components needed for a Phanta database: https://github.com/bhattlab/phanta/blob/main/databases.md

You're probably missing the species_genome_size.txt which can be calculated using the steps here (see Step 3: calculate genome size):
https://github.com/bhattlab/phanta/blob/main/databases.md

Hope this helps!

**Comment by gholizade** [comment]
Thank you for the info. Hi, when I run calculate_genome_size.py on my custom database (k2_viral_20231009), I get the error:                                   ValueError: No objects to concatenate
<img width="1612" height="277" alt="image" src="https://github.com/user-attachments/assets/147c5458-fc6c-4fd6-b874-77830c6a60b0" />

It seems that the script found zero FASTA/genome-files in the library/ folder. Acctully in library of k2_viral_20231009 only exist 
k2_viral_20231009/library/archaea/assembly_summary.txt (this was all I downloaded from kraken database). 
but in library_report.tsv there are  #Library	Sequence Name	URL
<img width="1591" height="607" alt="image" src="https://github.com/user-attachments/assets/3d1c186c-2db3-4f4e-b0b4-7aa647729992" />

Could you please le me know whether a species_genome_size.txt is generated only if FASTAs are present? or what files should be available
Thanks you.

**Reply by meenachakra** [reply]
Hello, yes, fasta files ending in .fna are required. Here is the code for the script: https://github.com/bhattlab/phanta/blob/main/pipeline_scripts/calculate_genome_size.py

Hope this helps.

**Comment by gholizade** [comment]
Hi, thank you for your support and fast reply. ok then. One more question. Now I have doubt if  k2_viral_20231009 database has the possibility to use for prophages detection using integrated_prophages_detector.py. Which files need to be check, names.dmp, nodes.dmp, or what else? and what I should expect to see on that files which make me sure I can do prophage detection? Thank you in advance.

**Reply by meenachakra** [reply]
As long as you've been able to successfully run Phanta, you should be able to run the prophage detector script! :)

**Reply by meenachakra** [reply]
Please see here for more description of the script's usage: https://github.com/bhattlab/phanta?tab=readme-ov-file#determine-which-detected-viruses-are-likely-integrated-prophages
